REGRESSION(279662@main): [ iOS 18 ] 6 TestWebKitAPI.UnifiedTextReplacement tests are constant failure

rr-codes opened the pull request:
#### 1bb6cdcc3e064f19e32c9e1eddc033be1613fe93
<pre>
REGRESSION(279662@main): [ iOS 18 ] 6 TestWebKitAPI.UnifiedTextReplacement tests are constant failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=275466">https://bugs.webkit.org/show_bug.cgi?id=275466</a>
<a href="https://rdar.apple.com/129560001">rdar://129560001</a>

Reviewed by Megan Gardner.

When receiving replacement text from Writing Tools in the Smart Reply case, a DocumentFragment is
created from the attributed string, which is then used in a `ReplaceSelectionCommand` with `MatchStyle::Yes`.

In this case, if there are internal style attributes involved, then an assertion failure will be reached.

Fix by removing this assertion, since it is perfectly valid to have internal attributes in this case,
and the case is handled fine already anyways.

* Source/WebCore/css/MutableStyleProperties.cpp:
(WebCore::MutableStyleProperties::setProperty):

Canonical link: <a href="https://commits.webkit.org/280000@main">https://commits.webkit.org/280000@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0925b566e00d0fd97e5bf7151e4309055768d1fb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55474 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34797 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7939 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58460 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5906 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57601 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42417 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6100 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44672 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4047 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57501 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32704 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47804 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25799 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29489 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5132 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4050 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51421 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5400 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60050 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30627 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5524 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52105 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31712 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47874 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51567 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12280 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32794 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31460 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
Nit: remove the braces around the if condition

rr-codes committed:
REGRESSION(279662@main): [ iOS 18 ] 6 TestWebKitAPI.UnifiedTextReplacement tests are constant failure
https://bugs.webkit.org/show_bug.cgi?id=275466
rdar://129560001

Reviewed by Megan Gardner.

When receiving replacement text from Writing Tools in the Smart Reply case, a DocumentFragment is
created from the attributed string, which is then used in a `ReplaceSelectionCommand` with `MatchStyle::Yes`.

In this case, if there are internal style attributes involved, then an assertion failure will be reached.

Fix by removing this assertion, since it is perfectly valid to have internal attributes in this case,
and the case is handled fine already anyways.

* Source/WebCore/css/MutableStyleProperties.cpp:
(WebCore::MutableStyleProperties::setProperty):

Canonical link: https://commits.webkit.org/280000@main
